infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/LogReplicationClientConfigListener.java

```diff
@@ -0,0 +1,147 @@
+import static org.corfudb.runtime.view.TableRegistry.CORFU_SYSTEM_NAMESPACE;
+
+
+@Slf4j
+public class LogReplicationClientConfigListener extends StreamListenerResumeOrFullSync {
+
+    /**
+     * TableSchema for fetching client registration table entries from CorfuStreamEntries
+     */
+    private static final TableSchema<ClientRegistrationId, ClientRegistrationInfo, Message> registrationTableSchema =
+            new TableSchema<>(LR_REGISTRATION_TABLE, ClientRegistrationId.class, ClientRegistrationInfo.class, Message.class);
+
+    /**
+     * TableSchema for fetching client metadata table entries from CorfuStreamEntries
+     */
+    private static final TableSchema<ClientDestinationInfoKey, DestinationInfoVal, Message> clientMetadataTableSchema =
+            new TableSchema<>(LR_MODEL_METADATA_TABLE, ClientDestinationInfoKey.class, DestinationInfoVal.class, Message.class);
+
+    /**
+     * Tables that this stream listener will be listening to.
+     */
+    private static final List<String> tablesOfInterest = new ArrayList<>(
+            Arrays.asList(LR_REGISTRATION_TABLE, LR_MODEL_METADATA_TABLE));
+
+    /**
+     * Accessing SessionManager for client register and unregister, which could lead to session creation and termination.
+     */
+    private final SessionManager sessionManager;
+
+    /**
+     * Accessing LogReplicationConfigManager to react accordingly when new entries found in client config tables
+     */
+    private final LogReplicationConfigManager configManager;
+
+    /**
+     * CorfuStore for subscribing this StreamListener
+     */
+    private final CorfuStore corfuStore;
+
+    public LogReplicationClientConfigListener(SessionManager sessionManager, LogReplicationConfigManager configManager, CorfuStore corfuStore) {
+        super(corfuStore, CORFU_SYSTEM_NAMESPACE, CLIENT_CONFIG_TAG, tablesOfInterest);
+        this.sessionManager = sessionManager;
+        this.configManager = configManager;
+        this.corfuStore = corfuStore;
+    }
+
+    /**
+     * Subscribe this stream listener to start monitoring the changes of LR client config tables.
+     */
+    public void start() {
+        CorfuStoreMetadata.Timestamp timestamp = configManager.preprocessAndGetTail();
+
+        log.info("Start log replication listener for client config tables from {}", timestamp);
+        corfuStore.subscribeListener(this, CORFU_SYSTEM_NAMESPACE, CLIENT_CONFIG_TAG, tablesOfInterest, timestamp);
+    }
+
+    /**
+     * A corfu update can/may have multiple updates belonging to different streams.
+     * This callback will return those updates as a list grouped by their Stream UUIDs.
+     * <p>
+     * Note: there is no order guarantee within the transaction boundaries.
+     *
+     * @param results is a map of stream UUID -> list of entries of this stream.
+     */
+    @Override
+    public void onNext(CorfuStreamEntries results) {
```

Comment:
onNext() would receive updates from 2 tables (registration and metadata) only? or additional tables can come up later?